### PR TITLE
add mathworks blog

### PR DESCRIPTION
**Pull request by pablogh-2000**

### CODE REVIEW

File: sections/blog/mathworks-klepsydra-integration.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+This snippet targets an object detection model optimized for speed.
+
+```matlab
+net = yolov3ObjectDetector('tiny-yolov3-coco');
+exportONNXNetwork(net.Network, 'yolov3tiny.onnx','OpsetVersion',12,'BatchSize',1);
+```
+*  `yolov3ObjectDetector` : Loads a pretrained "Tiny YOLOv3" model trained on the COCO dataset.
+*  `net.Network` : Extracts the underlying `DAGNetwork` or `dlnetwork` object from the detector wrapper.
+*  `OpsetVersion`, 12 : Specifies the ONNX operator set version to ensure compatibility with modern runtimes.
+*  `BatchSize`, 1 : Fixes the input size to process one image at a time, which is standard for real-time inference.
```
Comment (mruedac) [inline sections/blog/mathworks-klepsydra-integration.md]:
```suggestion
*  `BatchSize`, 1 : Fixes the input size to process one image at a time, which is standard for real-time inference. Must be fixed at 1.
```

File: sections/blog/mathworks-klepsydra-integration.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+This snippet targets a popular lightweight backbone used for image classification.
+
+```matlab
+net = ImagePretrainedNetwork('mobilenetv2');
+exportONNXNetwork(net, 'mobilenetv2.onnx','OpsetVersion',12,'BatchSize',1);
+```
+
+*  `ImagePretrainedNetwork` : A unified function to load the MobileNetV2 architecture.
+*  `exportONNXNetwork` : Converts the classification network directly to a `.onnx` file. Since MobileNetV2 is a classification network, we pass `net` directly rather than accessing a `.Network` property.
```
Comment (mruedac) [inline sections/blog/mathworks-klepsydra-integration.md]:
```suggestion
*  `exportONNXNetwork` : Converts the classification network directly to a `.onnx` file. Since MobileNetV2 is a `dlnetwork` object, we pass `net` directly rather than accessing a `.Network` property.
```

File: sections/blog/mathworks-klepsydra-integration.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+┡━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
+│ Add               │ 10             │ 10               │
+│ Clip              │ 35             │ 35               │
+│ Constant          │ 68             │ 68               │
+│ Conv              │ 52             │ 52               │
+│ GlobalAveragePool │ 1              │ 1                │
+│ MatMul            │ 1              │ 1                │
+│ Squeeze           │ 1              │ 1                │
+│ Model Size        │ 13.3MiB        │ 13.3MiB          │
+└───────────────────┴────────────────┴──────────────────┘
```
Comment (mruedac) [inline sections/blog/mathworks-klepsydra-integration.md]:
```suggestion
```bash
onnxsim onnxsim yolov3t
iny.onnx yolov3tiny_simp.onnx
Simplifying...
Finish! Here is the difference:
┏━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
┃                    ┃ Original Model ┃ Simplified Model ┃
┡━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
│ Add                │ 13             │ 0                │
│ BatchNormalization │ 11             │ 0                │
│ Cast               │ 39             │ 0                │
│ Ceil               │ 39             │ 0                │
│ Concat             │ 14             │ 1                │
│ Constant           │ 163            │ 28               │
│ Conv               │ 13             │ 13               │
│ Div                │ 39             │ 0                │
│ Floor              │ 26             │ 0                │
│ LeakyRelu          │ 11             │ 11               │
│ MaxPool            │ 6              │ 6                │
│ Mul                │ 13             │ 0                │
│ Pad                │ 13             │ 0                │
│ Relu               │ 13             │ 0                │
│ Resize             │ 1              │ 1                │
│ Shape              │ 13             │ 0                │
│ Split              │ 13             │ 0                │
│ Sub                │ 26             │ 0                │
│ Model Size         │ 33.8MiB        │ 33.8MiB          │
└────────────────────┴────────────────┴──────────────────┘
```

File: sections/blog/mathworks-klepsydra-integration.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+
+### Simplification
+
+After the network has been exported to ONNX format, it is recommended to use the ONNX Simplifier to simplify the ONNX graph and potentially reduce the size of the network.
+In case the network has quantized layers without shape information for the output layers, then the network will not be simplified efficiently.
+For the two example mobilenet models presented about we show how simplification affects the respective models.
+
+#### Simplifying model obtained from MATLAB
+
+For example, if model exported from MATLAB accepts 3 channel inputs of any dimensions, we define the input to have the same size. The command to run and the output obtained is presented below:
```
Comment (mruedac) [inline sections/blog/mathworks-klepsydra-integration.md]:
```suggestion
The command to run and the output obtained is presented below for yolov3tiny:
```

File: sections/blog/mathworks-klepsydra-integration.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+│ Squeeze           │ 1              │ 1                │
+│ Model Size        │ 13.3MiB        │ 13.3MiB          │
+└───────────────────┴────────────────┴──────────────────┘
+```
+
+### Klepsydra ONNX Requirements
+
+#### Static vs dynamic input size
+
+ONNX format supports networks with dynamic input size. However, as Klepsydra AI pre-allocates memory while loading the network and before any predictions are performed, it is essential that the expected input size of data is known. Setting the input size while creating the ONNX representation additionally allows redundant layers to be removed.
```
Comment (mruedac) [inline sections/blog/mathworks-klepsydra-integration.md]:
```suggestion
The models exported from Matlab used in this tutorial use a fixed input and output shapes. However, ONNX format in general, supports networks with dynamic input size. However, as Klepsydra AI pre-allocates memory while loading the network and before any predictions are performed, it is essential that the expected input size of data is known. Setting the input size while creating the ONNX representation additionally allows redundant layers to be removed. Static input size can be enforced by the simplifier adding the option `overwrite-input-shape`. For example in the previous case:

```bash
onnxsim --overwrite-input-shape="1,3,416,416" yolov3tiny.onnx yolov3tiny_fixed_input.onnx
```
```

File: sections/blog/mathworks-klepsydra-integration.md [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+The diagram below shows the deployment flow for a GR740 on RTEMS, showcasing the extended target support with Klepsydra AI integration.
+
+![Deployment GR740](/images/matlab_gr740_rtems_deployment.png)
+
+Further information on the deployment to GR740 processor can be found in the Klepsydra AI for GR740 guide.
+
+### The Integration Approach: ONNX Export
+
+This tutorial focuses on a seamless workflow between the training environment of MATLAB<sup>®</sup> and Klepsydra. The current approach to this integration is the  Open Neural Network Exchange (ONNX)  format. By exporting a model from MATLAB<sup>®</sup> using the `exportONNXNetwork` function, you can bridge the gap between MATLAB<sup>®</sup>’s powerful training environment and Klepsydra’s high-efficiency execution engine.
+The diagram below shows the deployment flow for a Raspberry PI4.
```
Comment (joseluisbf-kpsr) [inline sections/blog/mathworks-klepsydra-integration.md]:
```suggestion

The diagram below shows the deployment flow for a Raspberry PI4.
```
name Linux?

File: sections/blog/mathworks-klepsydra-integration.md [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+* *Computer Vision Toolbox Model for YOLO v3 Object Detection* (for the YOLOv3-Tiny model).
+* *Deep Learning Toolbox Model for MobileNet-v2 Network* (for the MobileNetV2 model).
+
+>  Note:  You can install these via the MATLAB<sup>®</sup>  Add-On Explorer  by searching for the names listed above.
+
+#### Klepsydra AI
+
+It is also required to have the  Klepsydra AI  software packages for your specific target platform. You can request a trial version by contacting [sales@klepsydra.com](mailto:sales@klepsydra.com).
+
+#### Google Protobufs
```
Comment (joseluisbf-kpsr) [inline sections/blog/mathworks-klepsydra-integration.md]:
```suggestion
#### Google Protobuf
```

Already included in the trial Docker?

Propose libprotobuf-dev DEB for APT ready distros?
(v3.19.0 mandatory?)

File: sections/blog/mathworks-klepsydra-integration.md [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+*  `OpsetVersion`, 12 : Specifies the ONNX operator set version to ensure compatibility with modern runtimes. Klepsydra current support is only for opset up to version 12.
+*  `BatchSize`, 1 : Fixes the input size to process one image at a time, which is standard for real-time inference. Klepsydra current support is only for batch size 1.
+
+#### 2. MobileNetV2 Export
+
+This snippet targets a popular lightweight backbone used for image classification.
+
+```matlab
+net = ImagePretrainedNetwork('mobilenetv2');
+exportONNXNetwork(net, 'mobilenetv2.onnx','OpsetVersion',12,'BatchSize',1);
```
Comment (joseluisbf-kpsr) [inline sections/blog/mathworks-klepsydra-integration.md]:
```suggestion
exportONNXNetwork(net, 'mobilenetv2.onnx', 'OpsetVersion', 12, 'BatchSize', 1);
```

File: sections/blog/mathworks-klepsydra-integration.md [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+
+This snippet targets an object detection model optimized for speed such as [YOLOV3](mathworks.com/help/vision/ref/yolov3objectdetector.html?s_tid=srchtitle_support_results_1_yolov3ObjectDetector)
+
+```matlab
+net = yolov3ObjectDetector('tiny-yolov3-coco');
+exportONNXNetwork(net.Network, 'yolov3tiny.onnx','OpsetVersion',12,'BatchSize',1);
+```
+*  `yolov3ObjectDetector` : Loads a pretrained "Tiny YOLOv3" model trained on the COCO dataset.
+*  `net.Network` : Extracts the underlying `DAGNetwork` or `dlnetwork` object from the detector wrapper.
+*  `OpsetVersion`, 12 : Specifies the ONNX operator set version to ensure compatibility with modern runtimes. Klepsydra current support is only for opset up to version 12.
```
Comment (joseluisbf-kpsr) [inline sections/blog/mathworks-klepsydra-integration.md]:
```suggestion
*  `'OpsetVersion', 12`: Specifies the ONNX operator set version to ensure compatibility with modern runtimes. Klepsydra current support is only for opset up to version 12.
```

File: sections/blog/mathworks-klepsydra-integration.md [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+
+### Klepsydra ONNX Requirements
+
+#### Static vs dynamic input size
+
+The models exported from MATLAB<sup>®</sup> used in this tutorial use a fixed input and output shapes. However, ONNX format in general, supports networks with dynamic input size. However, as Klepsydra AI pre-allocates memory while loading the network and before any predictions are performed, it is essential that the expected input size of data is known. Setting the input size while creating the ONNX representation additionally allows redundant layers to be removed. Static input size can be enforced by the simplifier adding the option `overwrite-input-shape`. For example in the previous case:
+
+```bash
+onnxsim --overwrite-input-shape="1,3,416,416" yolov3tiny.onnx yolov3tiny_fixed_input.onnx
+
```
Comment (joseluisbf-kpsr) [inline sections/blog/mathworks-klepsydra-integration.md]: closing ```

File: sections/blog/mathworks-klepsydra-integration.md [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+#### Batch Size
+
+Klepsydra AI does not do batched inference. So the batch size is always set to 1. In the ONNX representation this can simplify the network graph.
+
+#### Data Format : Channel first (NCHW) vs Channel last (NHWC)
+
+ONNX models represent data in the Channel first model. However, Klepsydra AI requires that data is in the Channel Last format (NHWC). As inputs and outputs are expected to be in channel last format, conversion of the DNN to ONNX may introduce Transpose layers in some cases. Klepsydra AI intelligently ignores these layers to avoid unnecessary data manipulation.
```
Comment (joseluisbf-kpsr) [inline sections/blog/mathworks-klepsydra-integration.md]: (change kpsr to the ONNX standard format? @mharshe)
